packages/webpack-extensive-lodash-replacement-plugin/index.js

```diff
@@ -332,74 +335,90 @@ const LODASH_MODULE_NAMES = [
 	'zipWith',
 ];
 
-function throwError( message, error = null ) {
-	throw new Error(
-		`[ExtensiveLodashReplacementPlugin] ${ message }${ error ? ' Error: ' : '' }${ error }`
+function createError( message, error = null ) {
+	return new Error(
+		`[ExtensiveLodashReplacementPlugin] ${ message }${ error ? ` Error: ${ error }` : '' }`
 	);
 }
 
-class ExtensiveLodashReplacementPlugin {
-	constructor( baseFile = './package.json' ) {
-		this.baseFile = path.resolve( baseFile );
-		let data;
+const resolverOptions = {
+	fileSystem: new CachedInputFileSystem( new NodeJsInputFileSystem(), 4000 ),
+	extensions: [ '.js' ],
+	resolveToContext: true,
+};
 
-		try {
-			data = fs.readFileSync( this.baseFile );
-		} catch ( error ) {
-			throwError( 'Could not read root package.json.', error );
-		}
+function getModuleForPath( rootPath, packageName ) {
+	const moduleResolver = ResolverFactory.createResolver( resolverOptions );
 
-		try {
-			const packageJson = JSON.parse( data );
-			this.lodashVersion =
-				packageJson && packageJson.dependencies && packageJson.dependencies[ 'lodash-es' ];
-		} catch ( error ) {
-			throwError( 'Could not parse root package.json.', error );
-		}
+	return new Promise( ( resolve, reject ) => {
+		moduleResolver.resolve( {}, rootPath, packageName, {}, ( error, filepath, context ) => {
+			if ( error ) {
+				reject(
+					createError(
+						`Could not find module ${ packageName } for import on ${ rootPath }.`,
+						error
+					)
+				);
+			}
 
-		if ( ! this.lodashVersion ) {
-			throwError( 'No lodash-es dependency in root package.json.' );
-		}
+			resolve( context );
+		} );
+	} );
+}
 
-		if ( ! semver.valid( this.lodashVersion ) ) {
-			throwError( `Invalid root package.json lodash-es version: ${ this.lodashVersion }.` );
-		}
+class ExtensiveLodashReplacementPlugin {
+	constructor( baseDir = '.' ) {
+		this.baseDir = path.resolve( baseDir );
+		this.baseLodashContext = getModuleForPath( this.baseDir, 'lodash' );
+		this.baseLodashESVersion = ( async () => {
+			const baseLodashES = await getModuleForPath( this.baseDir, 'lodash-es' );
+			return (
+				baseLodashES && baseLodashES.descriptionFileData && baseLodashES.descriptionFileData.version
+			);
+		} )();
```

Comment:
If you start using `nmf.getResolver( 'normal' )`, this initialization cannot be done in constructor. Plugin constructor is called when executing `webpack.config.js`. At that time, the compilation and other webpack objects like `nmf` don't exist yet.

Needs to be done lazily: `this.getBaseLodashContext( nmf )`

Reply:
Created an init function for this.